source/octf/cli/internal/param/ParamString.cpp

```diff
@@ -21,48 +21,64 @@ namespace cli {
 ParamString::ParamString()
         : Parameter()
         , m_values()
-        , m_defaultValue(1, "")
+        , m_defaultValues(1, "")
+        , m_hasDefaultValue(false)
         , m_maxLength(MAX_STRING_LENGTH)
-        , m_multipleValue(false) {}
+        , m_multipleValue(false)
+        , m_multipleValueLimit(0) {}
 
 void ParamString::setValue(CLIElement element) {
     if (!isValueSet()) {
-        string inputValue = element.getValue();
+        const auto &inputValue = element.getValue();
 
         if (inputValue.size() > m_maxLength) {
             throw InvalidParameterException("Value length of option '" +
                                             getLongKey() + "' is too long.");
         }
 
-        // Parse input for delimited multiple values
-        m_values = parseValuesToVector(inputValue);
-        setValueSet();
-
+        setValue(inputValue);
     } else {
         throw InvalidParameterException("Value of option '" + getLongKey() +
                                         "' is set multiple times.");
     }
 }
 
-const std::vector<std::string> &ParamString::getValue() const {
-    if (!isValueSet()) {
-        return m_defaultValue;
+const std::string &ParamString::getValue() const {
+    if (!isMultipleValue()) {
+        if (isValueSet()) {
+            return m_values[0];
+        } else {
+            return m_defaultValues[0];
+        }
     } else {
-        return m_values;
+        throw InvalidParameterException("Option '" + getLongKey() +
+                                        "' access error, use multiple getter");
     }
 }
 
-void ParamString::setValue(std::string value) {
-    m_values = parseValuesToVector(value);
-    setValueSet();
+const std::vector<std::string> &ParamString::getMultipleValue() const {
+    if (isMultipleValue()) {
+        if (isValueSet()) {
+            return m_values;
+        } else {
+            return m_defaultValues;
+        }
+    } else {
+        throw InvalidParameterException("Option '" + getLongKey() +
+                                        "' access error, use regural getter");
+    }
 }
 
-bool ParamString::isValid(const std::string &value) {
-    if (value.size() > MAX_STRING_LENGTH) {
-        return false;
+void ParamString::setValue(std::string value) {
```

Comment:
I think the input can be const - I believe all instances where calling parseValuesToVector should have the string parameter as const.